src/drv/stm32cube/bsp_i2c_master.c

```diff
@@ -192,6 +200,40 @@ bsp_status_t bsp_i2c_stop(bsp_dev_i2c_t dev_num)
 	return BSP_OK;
 }
 
+/** \brief Set SCL to float and wait for slave device to be ready
+ */
+static void i2c_master_set_scl_float_and_wait_ready(void)
```

Comment:
Add a return value **bsp_status_t** BSP_OK or an error in case of Timeout BSP_TIMEOUT

Reply:
Sounds good! and this is the result after the update. 

The error message is shown only when clock stretch is enabled:
```
i2c1> clock-stretch 1
i2c1> show
GPIO resistor: pull-up
Mode: master
Frequency: 100khz (50khz, 400khz, 1mhz)
Clock stretch timeout: 1 ticks / 10.00 us (0 = Disabled)
i2c1> [w 0x50] &:10 [w 0x50 0x00] &:10 [0x51 r:1]
I2C START
WRITE: 0x50 ACK
i2c clock stretching timed out. Please consider increase the timeout with clock-stretch command to see if it helps.
WRITE error:3
I2C STOP
I2C START
WRITE: 0x50 NACK 0x00 NACK
i2c clock stretching timed out. Please consider increase the timeout with clock-stretch command to see if it helps.
WRITE error:3
I2C STOP
I2C START
WRITE: 0x51 ACK
i2c clock stretching timed out. Please consider increase the timeout with clock-stretch command to see if it helps.
WRITE error:3
READ: 0x07
i2c clock stretching timed out. Please consider increase the timeout with clock-stretch command to see if it helps.
READ error:3
I2C STOP
```

When it is disabled, it does the same as what we are doing now:
```
i2c
GPIO resistor: pull-up
Mode: master
Frequency: 100khz (50khz, 400khz, 1mhz)
Clock stretch timeout: 0 ticks / 0.00 us (0 = Disabled)
i2c1> [w 0x50] &:10 [w 0x50 0x00] &:10 [0x51 r:1]
I2C START
WRITE: 0x50 ACK
I2C STOP
I2C START
WRITE: 0x50 NACK 0x00 ACK
I2C STOP
I2C START
WRITE: 0x51 ACK
READ: 0x07 NACK
I2C STOP
```

Reply:
i have made a bit change to handle the error inside i2c mode in a unified place, hence they are tracked together now. resolved as dup for now, so we have a single place for discussion.

Reply:
1 small note: I add a function to print the i2c only error inside i2c mode. I am aware of hydrabus_mode.c has code to print out the error, but I feel that is way to generic and probably not a good idea to put mode and error checking code there, especially, BSP_TIMEOUT could mean different things, e.g. clock stretch timeout in IO, while no data returned for sniff. So maybe within i2c mode is a better place.